Source/WebCore/platform/graphics/cg/UTIRegistry.cpp

```diff
@@ -107,14 +107,38 @@ MemoryCompactRobinHoodHashSet<String>& additionalSupportedImageTypes()
     return additionalSupportedImageTypes;
 }
 
+struct ImageTypeMimeTypePair {
+    ASCIILiteral imageType;
+    ASCIILiteral mimeType;
+};
+
+static const HashMap<String, String, ASCIICaseInsensitiveHash>& additionalImageTypeMimeTypeMap()
+{
+    static NeverDestroyed<HashMap<String, String, ASCIICaseInsensitiveHash>> mimeTypesMap = [] {
+        HashMap<String, String, ASCIICaseInsensitiveHash> map;
+        static constexpr ImageTypeMimeTypePair additionalPairs[] = {
+            { "com.adobe.photoshop-image"_s, "application/x-photoshop"_s },
+        };
+        for (auto& [imageType, mimeType] : additionalPairs)
+            map.add(imageType, mimeType);
+        return map;
+    }();
+    return mimeTypesMap;
+}
+
```

Comment:
I think it is a very rare scenario which requires supporting more than one MIMEType for a given UTI. And I would not expect that we will be adding entries to this HashMap at least in the near future. So I would suggest just adding this as a single if-statement:

```
if (equalLettersIgnoringASCIICase(imageType, "com.adobe.photoshop-image"_s)
    return "application/x-photoshop"_s;
```